jdaviz/components/toolbar_nested.py

```diff
@@ -53,13 +65,74 @@ def __init__(self, viewer, tools_nested, default_tool_priority=[]):
         self.default_tool_priority = default_tool_priority
         self._handle_default_tool()
 
+        for msg in (AddDataMessage, RemoveDataMessage, ViewerAddedMessage, ViewerRemovedMessage,
+                    SpectralMarksChangedMessage):
+            self.viewer.hub.subscribe(self, msg,
+                                      handler=lambda _: self._update_tool_visibilities())
```

Comment:
this part isn't terribly extensible for downstream... we might eventually want to consider the tools storing `tool.toolbar`, subscribing themselves, and being able to call `_update_tool_visibilities`, but let's defer that until there is an actual use-case.